Automattic/jetpack
# VideoPress: Refactor loading state on video details page and disable redirect on save

Fixes #29238
Fixes #29237

## Proposed changes:

* Refactors the UI to separate loading state from other states when the interface should be disabled
* Removes the redirection to the dashboard when updating a video
* Adds the `filename` value.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the VideoPress dashboard and add some videos
* Go to the details page of a video and save some changes
* Check that the UI no longer blocks the view of the data
![2023-03-08_15-57-48](https://user-images.githubusercontent.com/8486249/223809297-e2696480-eea1-467b-9f2a-1d48f941c5bf.png)
* Check that there is no redirection to the dashboard
* Click on `Go back`
* Check that there is no modal prompting the user to save before leaving
* Go back to the details page
* Check that the file name is displayed.
* Delete the video
* Check that the UI goes to the same state as when updating
* Check that you are redirected to the dashboard after deleting the video
